apps/omg_utils/lib/omg_utils/metrics.ex

```diff
@@ -40,18 +40,61 @@ defmodule OMG.Utils.Metrics do
   def measure_start(body, context) do
     # NOTE: the namespace and event group naming convention here is tentative.
     # It is possible we'll revert to standard coarser division into `web` and `background` namespaces Appsignal suggests
-    namespace = context.module |> Module.split() |> List.last() |> String.to_existing_atom()
-
-    if Enum.find(@discard_namespace_metrics, &match?(^&1, namespace)),
+    if Enum.find(@discard_namespace_metrics, &match?(^&1, namespace(context))),
       do: body,
-      else: Appsignal.Instrumentation.Decorators.transaction(namespace, body, context)
+      else: start_trace(body, context)
   end
 
   def measure_event(body, context) do
-    event_group = context.module |> Module.split() |> List.last() |> String.to_existing_atom()
-
-    if Enum.find(@discard_namespace_metrics, &match?(^&1, event_group)),
+    if Enum.find(@discard_namespace_metrics, &match?(^&1, namespace(context))),
       do: body,
-      else: Appsignal.Instrumentation.Decorators.transaction_event(event_group, body, context)
+      else: start_span(body, context)
+  end
+
+  defp start_trace(body, context) do
```

Comment:
hm... is this the recommended way of building traces? why do we need to write these macros ourselves?